src/contexts/AuthContext.js

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, useContext } from 'react';
+import { auth } from '../firebase';
+import { 
+  signInWithPopup, 
+  GoogleAuthProvider, 
+  signOut as firebaseSignOut 
+} from 'firebase/auth';
+import { useAuthState } from 'react-firebase-hooks/auth';
+
+const AuthContext = createContext();
+
+export function useAuth() {
+  return useContext(AuthContext);
+}
+
+export function AuthProvider({ children }) {
+  const [user, loading, error] = useAuthState(auth);
+
+  const signInWithGoogle = async () => {
+    const provider = new GoogleAuthProvider();
+    try {
+      const result = await signInWithPopup(auth, provider);
+      
+      // Verifica se o email termina com @kodus.io
+      if (!result.user.email.endsWith('@kodus.io')) {
+        // Se não for um email válido, faz logout e mostra erro
+        await firebaseSignOut(auth);
+        alert('Apenas emails @kodus.io são permitidos');
+        return;
+      }
+      
+    } catch (error) {
+      console.error('Error signing in with Google:', error);
+    }
+  };
+
+  const signOut = () => {
+    firebaseSignOut(auth);
+  };
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const signOut = async () => {
    try {
      await firebaseSignOut(auth);
    } catch (error) {
      console.error('Error signing out:', error);
      throw error;
    }
  };
```
The signOut function should handle potential errors that might occur during the sign-out process. Currently, it doesn't have any error handling.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;